var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -174,4 +195,10 @@ def cmake_args(self):
         else:
             options.append('-DGMX_RELAXED_DOUBLE_PRECISION:BOOL=OFF')
 
+        fft_value = self.spec.variants['fft'].value
+        if fft_value == 'fftw3':
+            options.append('-DGMX_FFT_LIBRARY=fftw3')
+        elif fft_value == 'mkl':
+            options.append('-DGMX_FFT_LIBRARY=mkl')
+
```

Comment:
```suggestion
        if '^fftw@3:' in self.spec:
            options.append('-DGMX_FFT_LIBRARY=fftw3')
        elif '^intel-mkl' in self.spec:
            options.append('-DGMX_FFT_LIBRARY=mkl')

```

Reply:
Thanks. Your solution is much simpler, although technically it is two different interfaces that are used by gromacs (there is actually a third - fftpack, which is builtin). Furthermore, for cuda builds neither fftw nor mkl seems to be needed http://manual.gromacs.org/documentation/current/install-guide/index.html#fast-fourier-transform-library , but I haven't used or tried to compile it . I will put in your changes and use a `when='~cuda'` for the dependency. Maybe the maintainers can comment on that.

Reply:
I had to change the suggestions a bit such that other `mkl` providers also work. 
Tested with intel@19.0.4 and gcc@7.5.0.